.github/workflows/docs-ci.yml

```
@@ -16,23 +26,29 @@ jobs:
       GITHUB_BRANCH_NAME: ${{ github.ref }}
 
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Set up JDK 8
-        uses: actions/setup-java@v1
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
         with:
+          distribution: temurin
           java-version: 8
-
       - name: Build docs with Gradle
         uses: gradle/gradle-build-action@v2
         with:
           arguments: --stacktrace --info dokkaHtmlMultiModule
-
-      - name: Push docs to gh-pages
-        uses: JamesIves/github-pages-deploy-action@v4
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v1
         with:
-          folder: build/dokka/htmlMultiModule
-          branch: gh-pages
-          git-config-name: GitHub Actions
-          git-config-email: actions@github.com
-          commit-message: Update docs
+          path: build/dokka/htmlMultiModule
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: build
```

```suggestion
    needs: update_docs
```
`update_docs` could instead be renamed